### PR TITLE
Hot wallet: Generate and register a new algorand wallet

### PR DESCRIPTION
**Pull request by Riddlerrr**
Task: https://www.pivotaltracker.com/story/show/176550109

Check if wallet keys stored in Redis:
- *YES* do nothing
- *NO* Generate new keys, register it on Comakery by API and if it was registered successfully then store it into Redis

### CODE REVIEW

File: hotwallet/.env.required [hunk 1]
```diff
@@ -0,0 +1,5 @@
+PROJECT_ID=5
```
Comment (aquabu) [inline hotwallet/.env.required:1]: seems like this should be a `.env.example` file and that the actual values shouldn't be checked in?

File: hotwallet/app.js [hunk 2]
```diff
@@ -1 +1,87 @@
...
+
+function generateAlgorandKeyPair() {
+  const account = algosdk.generateAccount()
+  const mnemonic = algosdk.secretKeyToMnemonic(account.sk);
+  const new_wallet = { address: account.addr, mnemonic: mnemonic }
+
+  return new_wallet
+}
+
+function isAllVariablesSet() {
```
Comment (aquabu) [inline hotwallet/app.js:24]: rename to `areAllVariablesSet()` or `checkAllVariablesAreSet()`

File: hotwallet/app.js [hunk 3]
```diff
@@ -1 +1,87 @@
-// It is a very beginning of the Hotwallet
+require('dotenv').config()
```
Comment (aquabu) [inline hotwallet/app.js:1]: This is critical code so it needs testing - probably in Jest.

**Review by aquabu — CHANGES_REQUESTED** [review]

**Review by aquabu — APPROVED** [review]

File: hotwallet/lib/hotwalletUtils.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    Boolean(envs.purestakeApi) && Boolean(envs.redisUrl)
+}
+
+exports.setRedisErrorHandler = function setRedisErrorHandler(redisClient) {
+  redisClient.on("error", function (err) {
+    console.error(`Redis client error: ${err}`);
+  })
+}
+
+exports.generateAlgorandKeyPair = function generateAlgorandKeyPair() {
```
Comment (Alexey1100) [inline hotwallet/lib/hotwalletUtils.js:14]: Seems like duplicate of `hotwallet/app.js:18`

**Review by aquabu — APPROVED** [review]

**Pull request merged**